packages/engine/src/common/functions/traverse.ts

```diff
@@ -0,0 +1,18 @@
+import { Object3D } from 'three'
+
+/**
+ * Object3D traverse function with abort ability
+ * @param object
+ * @param callback
+ * @returns
+ */
+export function traverse(object: Object3D, callback) {
```

Comment:
let's make this more descriptive by renaming it `traverseUntil` or something

Reply:
This is a temporary file. There is another file (Object3DUtils.ts) in Hands IK
 #5549 PR. I will integrate this function into that file once merged.